src/document-converter.ts

```diff
@@ -269,6 +300,38 @@ export class DocumentConverter {
     return [...document.getFeatures({kind: 'html-import'})];
   }
 
+  /**
+   * Finds the comment containing the FileConversionSettings object and returns
+   * the parsed object. Throws if more than one comment could be parsed as the
+   * FileConversionSettings object.
+   */
+  private getFileConversionSettings(document: ParsedHtmlDocument):
+      Partial<FileConversionSettings> {
+    const treeAdapter = parse5.treeAdapters.default;
+
+    let settings: Partial<FileConversionSettings>|null = null;
+    document.visit([(node: parse5.ASTNode) => {
```

Comment:
I think visiting every node here is what's slowing everything down. I'm going to look around in the analyzer to see if it already found these comments.